### PR TITLE
Add DEStat

### PR DESCRIPTION
**Pull request by YingboMa**
Ref: https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/667

### CODE REVIEW

File: src/diffeqfunction.jl [hunk 1]
```diff
@@ -186,7 +202,7 @@ function ODEFunction{iip,true}(f;
                  typeof(jac),typeof(jac_prototype),typeof(invW),typeof(invW_t),
                  typeof(paramjac),typeof(syms)}(
                  f,mass_matrix,analytic,tgrad,jac,jac_prototype,invW,invW_t,
-                 paramjac,syms)
+                 paramjac,syms,Ref(Int(1)))
```
Comment (ChrisRackauckas) [inline src/diffeqfunction.jl]: why 1?

Comment (YingboMa) [reply src/diffeqfunction.jl]: Oops, that is a mistake.

File: src/destats.jl [hunk 2]
```diff
@@ -0,0 +1,20 @@
+mutable struct DEStat{E}
```
Comment (ChrisRackauckas) [inline src/destats.jl]: DEStats

File: src/DiffEqBase.jl [hunk 3]
```diff
@@ -117,6 +117,7 @@ include("solutions/rode_solutions.jl")
 include("solutions/dae_solutions.jl")
 include("solutions/monte_solutions.jl")
 include("solutions/solution_interface.jl")
+include("DEStats.jl")
```
Comment (eljungsk) [inline src/DiffEqBase.jl]: Hi,

This line unfortunately breaks this package on Linux due to case sensitivity.

**Pull request merged**